File: Pattern31.go
// *       * 
//  *     *  
//   *   *   
//    * *    
//     *     
//    * *    
//   *   *   
//  *     *  
// *       * 

package main

import "fmt"

func main() {

	var n int
	fmt.Printf("Enter the row number:")
	fmt.Scanln(&n)
	var flag int
	for i := 1; i <= 2*n-1; i++ {
		if i <= n {
			flag = i
		} else {
			flag = 2*n - i
		}
		var flag1 int = n
		for k := 1; k <= flag; k++ {
			fmt.Printf(" ")
		}
		for j := flag1; j >= flag; j-- {
			if j == flag1 || j == flag {
				fmt.Printf("* ")
			} else {
				fmt.Printf("  ")
			}

		}
		fmt.Printf("\n")
	}
}

File: Pattern31.py

# *       * 
#  *     *  
#   *   *   
#    * *    
#     *     
#    * *    
#   *   *   
#  *     *  
# *       * 

n = int(input("enter the row number:"))
for i in range(1,2*n):
    if( i <= n):
        flag = i
    else :
        flag = 2*n - i
    flag1 = n
    for k in range(1,flag+1):
        print(" ",end='')
    for j in range(flag1,flag-1,-1):
        if (j == flag1 or j == flag):
           print("* ",end='')
        else:
            print("  ",end='')
    print()

